### PR TITLE
Top action bar and bottom action bar

### PR DESCRIPTION
**Pull request by MelissaWu-SS**
Add TopActionBar and BottomActionBar Components to LeftAndMain.

Screen Shot 2022-02-23 at 2 00 06 PM

### CODE REVIEW

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
I'm thinking maybe, we want to be able to just pass a list of actions as an array and let our component figure out how to render that. The problem is the more flexibility you give people the more they have a chance to screw it up. I'm thinking something like this:
```jsx
const topActions: [
  {label: "Save", icon: "save", style: "primary", onClick: doSomething},
  ...
];

return <LeftAndMain topActions={topActions}>...</LeftAndMain>
```

But I also see how being able to pass a react component directly into LeftAndMain could be helpful for more advance use case

File: client/src/components/ActionBar.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+const ActionBar = ({actions, isBottomActionBar}) => {
+  const actionBarClass = classNames({
+    "action-bar": true,
+    "fill-width": true,
+    "action-bar--bottom": isBottomActionBar
+  });
+
+  const actionButtons = actions.map((action) => (
+      <Button {...action}>
```
Comment (maxime-rainville) [inline client/src/components/ActionBar.js:22]: I think this thing will throw a warning if you don't give a unique `key` prop to each button.

File: client/src/components/LeftAndMain.js [hunk 2]
```diff
@@ -9,15 +11,36 @@ const breadcrumbs = [
...
+ * @constructor
+ */
+const LeftAndMain = ({children, topActions, bottomActions}) => {
   return (
-    <div className="fill-height">
+    <div className="left-and-main fill-height">
       <Toolbar className="fill-width">
         <Breadcrumb multiline crumbs={breadcrumbs} />
       </Toolbar>
+      { topActions.length > 0 ? <ActionBar actions={topActions} /> : '' }
```
Comment (maxime-rainville) [inline client/src/components/LeftAndMain.js:29]:
```suggestion
      { topActions.length > 0 && <ActionBar actions={topActions} /> }
```

File: client/src/components/LeftAndMain.js [hunk 3]
```diff
@@ -9,15 +11,36 @@ const breadcrumbs = [
...
       {children}
+      { bottomActions.length > 0 ? <ActionBar actions={bottomActions} isBottomActionBar /> : '' }
```
Comment (maxime-rainville) [inline client/src/components/LeftAndMain.js:31]:
```suggestion
      { bottomActions.length > 0 && <ActionBar actions={bottomActions} isBottomActionBar /> }
```

File: client/src/components/tests/LeftAndMain-story.js [hunk 4]
```diff
@@ -5,19 +5,32 @@ import { withKnobs, text, boolean, select } from '@storybook/addon-knobs/react';
...
 setAddon(JSXAddon);
 
+const topActions = [
+  { label: "Save", color: "primary", size: 'md'},
+  { label: "Delete", color: "secondary", size: 'md'},
+];
+
+const bottomActions = [
+  { label: "Edit", icon: "edit", color: "primary", size: 'md', outline: true },
+];
```
Comment (maxime-rainville) [inline client/src/components/tests/LeftAndMain-story.js:21]: I would discourage people from using `size` here.

```suggestion
const topActions = [
  { label: "Save", color: "primary"},
  { label: "Delete", color: "secondary"},
];

const bottomActions = [
  { label: "Edit", icon: "edit", color: "primary", outline: true },
];
```

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Code looks great ... I left a few very minor suggestions. I'll try running it locally now.

**Comment by maxime-rainville** [comment]
If you have lots of content, it will be displayed under the bottom card. The bottom bar probably should have a solid background colour with a border ... and we need to make sure the main area as some sort of margin so it can always clear the bottom bar when its present.

Maybe we leave this for a polishing phase later.


image

File: client/src/components/ActionBar.js (hunk 1 above)
Comment (MelissaWu-SS) [reply client/src/components/ActionBar.js]: Ah, yeah, I should make the `key` prop to be required. Thanks!

**Comment by MelissaWu-SS** [comment]
> Code looks great ... I left a few very minor suggestions. I'll try running it locally now.

@maxime-rainville Thanks for the suggestions, I've made the changes, ready for another PR😃

**Review by maxime-rainville — APPROVED** [review]

**Pull request merged**